### PR TITLE
Add ReadKey to IConsole

### LINKED ISSUE #68: Console.ReadKey() implementation for SystemConsole
Generally I like the analyzer's warnings about usage of `System.Console`. However, in case of `System.Console.ReadKey()` there's no corresponding method in `IConsole`.

Either it may be added to `IConsole`, or the analyzer should stop worrying about `ReadKey()`.

### PR DESCRIPTION
**Pull request by alexrosenfeld10**
Closes #68

### CODE REVIEW

File: CliFx/Infrastructure/IConsole.cs [hunk 1]
```diff
@@ -87,6 +87,12 @@ public interface IConsole
         /// Clears the console buffer and corresponding console window of display information.
         /// </summary>
         void Clear();
+
+        /// <summary>
+        /// Wrapper to the <a href="https://docs.microsoft.com/en-us/dotnet/api/system.console.readkey">System.Console.ReadKey</a> method
+        /// </summary>
+        /// <param name="intercept"></param>
```
Comment (Tyrrrz) [inline CliFx/Infrastructure/IConsole.cs:94]:
```suggestion
        /// <summary>
        /// Obtains the next character or function key pressed by the user.
        /// </summary>
```

File: CliFx/Infrastructure/IConsole.cs [hunk 2]
```diff
@@ -87,6 +87,12 @@ public interface IConsole
...
+        void ReadKey(bool intercept);
```
Comment (Tyrrrz) [inline CliFx/Infrastructure/IConsole.cs:95]:
```suggestion
        void ReadKey(bool intercept = false);
```

Comment (alexrosenfeld10) [reply CliFx/Infrastructure/IConsole.cs:95]: duh. my bad, thanks.

File: CliFx/Infrastructure/FakeConsole.cs [hunk 3]
```diff
@@ -84,12 +84,17 @@ public void RequestCancellation(TimeSpan? delay = null)
...
         }
-        
+
         /// <inheritdoc />
         public void Clear()
         {
         }
 
+        /// <inheritdoc />
+        public void ReadKey(bool intercept)
```
Comment (Tyrrrz) [inline CliFx/Infrastructure/FakeConsole.cs:94]:
```suggestion
        public void ReadKey(bool intercept = false)
```

File: CliFx/Infrastructure/SystemConsole.cs [hunk 4]
```diff
@@ -89,10 +89,13 @@ public CancellationToken RegisterCancellationHandler()
 
             return (_cancellationTokenSource = cts).Token;
         }
-        
+
         /// <inheritdoc />
         public void Clear() => Console.Clear();
 
+        /// <inheritdoc />
+        public void ReadKey(bool intercept) => Console.ReadKey(intercept);
```
Comment (Tyrrrz) [inline CliFx/Infrastructure/SystemConsole.cs:97]:
```suggestion
        public void ReadKey(bool intercept = false) => Console.ReadKey(intercept);
```

**Comment by Tyrrrz** [comment]
Sorry, I accidentally submitted the suggestions individually instead of together.
Looks good otherwise :+1:

**Comment by alexrosenfeld10** [comment]
Some obvious things I missed here - thanks for the eyes.

**Comment by alexrosenfeld10** [comment]
No problem, all updated now

**Comment by Tyrrrz** [comment]
Thanks!

**Pull request merged**